components/script/dom/htmlmediaelement.rs

```diff
@@ -1220,6 +1234,118 @@ impl HTMLMediaElement {
             },
             PlayerEvent::MetadataUpdated(ref metadata) => {
                 // https://html.spec.whatwg.org/multipage/#media-data-processing-steps-list
+                // => If the media resource is found to have an audio track
+                if !metadata.audio_tracks.is_empty() {
+                    for (i, _track) in metadata.audio_tracks.iter().enumerate() {
+                        // Step 1.
+                        let kind = match i {
+                            0 => "main",
+                            _ => "",
+                        };
+                        let window = window_from_node(self);
+                        let audio_track = AudioTrack::new(
+                            &window,
+                            DOMString::new(),
+                            DOMString::from(kind),
+                            DOMString::new(),
+                            DOMString::new(),
+                        );
+
+                        // Step 2.
+                        self.AudioTracks().add(&audio_track);
+
+                        // Step 3.
+                        audio_track.set_enabled(false);
+
+                        // Step 4
+                        // https://www.w3.org/TR/media-frags/#media-fragment-syntax
+                        // https://github.com/servo/servo/issues/22366
+
+                        // Step 5. & 6,
+                        if audio_track.enabled() || self.AudioTracks().enabled_index().is_none() {
+                            let window = window_from_node(self);
+                            let task_source = window.task_manager().media_element_task_source();
+                            let this = Trusted::new(self);
+
+                            task_source
+                                .queue(
+                                    task!(media_track_change: move || {
+                                        let this = this.root();
+                                        this.AudioTracks().set_enabled(0, true);
+                                        this.upcast::<EventTarget>().fire_event(atom!("change"));
+                                    }),
+                                    window.upcast(),
+                                )
+                                .unwrap();
+                        }
+
+                        // Steps 7.
+                        TrackEvent::new(
```

Comment:
This creates an event object but does not do anything with it. We should store it in a variable and use the [Event::fire](https://doc.servo.org/script/dom/event/struct.Event.html#method.fire) API to dispatch it at the track list.